lib/core/src/animation/fixed_step_animator.rs

```diff
@@ -0,0 +1,48 @@
+use super::DifferentialTimeAnimator;
+use super::FnAnimation;
+
+use nalgebra::zero;
+
+// =============================
+// === FixedStepAnimatorData ===
+// =============================
+
+struct FixedStepAnimatorData {
+    closure          : Box<dyn FnMut(f32)>,
+    step_duration    : f32,
+    accumulated_time : f32
+}
+
+impl FixedStepAnimatorData {
+    pub fn new<F:FnAnimation>(steps_per_second:f32, f:F) -> Self {
+        let closure          = Box::new(f);
+        let step_duration    = 1000.0 / steps_per_second;
+        let accumulated_time = zero();
+        Self { closure,step_duration,accumulated_time }
+    }
+}
+
+
+
+// ================
+// === Animator ===
+// ================
+
+/// This structure is aimed to run a closure at a fixed time rate.
+pub struct FixedStepAnimator {
+    _differential_animator: DifferentialTimeAnimator
+}
+
+impl FixedStepAnimator {
+    pub fn new<F:FnAnimation>(steps_per_second:f32, f:F) -> Self {
+        let mut data               = FixedStepAnimatorData::new(steps_per_second, f);
+        let _differential_animator = DifferentialTimeAnimator::new(move |delta_time| {
+            data.accumulated_time += delta_time;
+            while data.accumulated_time > data.step_duration {
+                data.accumulated_time -= data.step_duration;
+                (data.closure)(data.step_duration);
```

Comment:
I still think this is bad that you pass to the inner closure always `1 / step_duration`. It would be much more sensible to pass there a **real time that passed between frames**. Now you are repeating information. In order to create `FixedStepAnimator` you need to provide `steps_per_second` and a function. Why then the first argument to this function is always `1 / steps_per_second`? It is redundant information and you should never pass reduntant information around.

Reply:
Keeping the input of the closure as `step_duration` even if it's taking more than `step_duration` is an useful assumption. But let me also discuss about the design:

If `step_duration = 1s` and the frame actually takes `2.5s`, it will call the closure twice with `step_duration = 1s` and leave the remainder `0.5s` to be accumulated.

We could run it a third time with `0.5s`, but the remainder value would be variable and it isn't useful for fixed step physics.

Analyzing the above situation further, we can see that this could result in 4 iterations in `3s`, instead of 3 iterations in `3s`, if the first frame takes `2.5s` (two `1s` calls and one `0.5s` call) and the second one takes `0.5s` (just one `0.5s` call).

Reply:
If step durration is `1s` and frame took `2.5s` you should finally run 3 frames and use the interpolation to display the correct value between `2s` and `3s`. 

I completely dont feel this point: `Keeping the input of the closure as step_duration even if it's taking more than step_duration is an useful assumption.` - why? You are providing this information when creating the `FixedStepAnimator`. Why you need to provide to the closure the same information that you provided when defining the closure?